src/rust/engine/fs/store/src/lib.rs

```diff
@@ -969,32 +1042,43 @@ impl Store {
                 *missing_file_digests.iter().next().unwrap(),
             )
         })?;
-        let _ = future::try_join_all(missing_file_digests.into_iter().map(
-            |file_digest| async move {
-                if let Err(e) = remote
-                    .download_digest_to_local(
-                        self.local.clone(),
-                        file_digest,
-                        EntryType::File,
-                        None,
-                    )
-                    .await
-                {
-                    log::debug!("Missing file digest from remote store: {:?}", file_digest);
-                    in_workunit!(
-                        "missing_file_counter",
-                        Level::Trace,
-                        |workunit| async move {
-                            workunit.increment_counter(Metric::RemoteStoreMissingDigest, 1);
-                        },
-                    )
-                    .await;
-                    return Err(e);
-                }
-                Ok(())
-            },
-        ))
-        .await?;
+
+        if remote.store.batch_load_supported() {
```

Comment:
all the plumbing of `batch_load_supported` is because of this. I need to check if the remote supports batching before using it here